dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassDumper.java

```diff
@@ -0,0 +1,180 @@
+      int nameAndTypeIndex = constantMethodref.getNameAndTypeIndex();
+      Constant constantNameAndTypeRaw = constantPool.getConstant(nameAndTypeIndex);
+      if (!(constantNameAndTypeRaw instanceof ConstantNameAndType)) {
+        throw new RuntimeException(
+            "Failed to lookup nameAndType constant indexed " + nameAndTypeIndex);
+      }
+      ConstantNameAndType constantNameAndType = (ConstantNameAndType) constantNameAndTypeRaw;
+      String methodName = constantNameAndType.getName(constantPool);
+      String signature = constantNameAndType.getSignature(constantPool);
+      ConstantPoolMethodref methodref = new ConstantPoolMethodref(className, methodName, signature);
+      constantPoolMethodrefs.add(methodref);
+    }
+    return constantPoolMethodrefs;
+  }
+
+  /**
+   *  Lists all methodref entries defined in ConstantPool of class file
+   *
+   * @param classFileStream stream of a class file
+   * @param fileName name of the file that contains class
+   * @return List of all methodref entries defined in ConstantPool
+   * @throws IOException when there is problem in reading classFileStream
+   */
+  @VisibleForTesting
+  static List<ConstantPoolMethodref> listConstantPoolMethodref(InputStream classFileStream,
+      String fileName) throws IOException {
+    ClassParser parser = new ClassParser(classFileStream, fileName);
+    JavaClass javaClass = parser.parse();
+    return listConstantPoolMethodRef(javaClass);
+  }
+
+  /**
+   *  Lists the methodref entries owned by the defined class in class file
+   *
+   * @param classFileStream stream of a class file
+   * @param fileName name of the file that contains class
+   * @return methodref entries owned by the class
+   * @throws IOException when there is problem in reading classFileStream
+   */
+  public static List<ConstantPoolMethodref> listOwningConstantPoolMethodref(InputStream classFileStream,
+      String fileName) throws IOException {
+    ClassParser parser = new ClassParser(classFileStream, fileName);
+    JavaClass javaClass = parser.parse();
+    List<ConstantPoolMethodref> methodrefs = listConstantPoolMethodRef(javaClass);
+    Set<String> owningClass = Sets.newHashSet(javaClass.getClassName());
+    List<ConstantPoolMethodref> owningMethodrefs =  methodrefs
+            .stream()
+            .filter(methodref -> owningClass.contains(methodref.getClassName()))
+            .collect(Collectors.toList());
+    return owningMethodrefs;
+  }
+
+  /**
+   *  Lists the methodref entries not owned by the defined class in class file
+   *  The list the class attemts to invoke the methods defined elsewhere
+   *
+   * @param classFileStream stream of a class file
+   * @param fileName name of the file that contains class
+   * @return methodref entries external to the class
+   * @throws IOException when there is problem in reading classFileStream
+   */
+  public static List<ConstantPoolMethodref> listExternalConstantPoolMethodref(
```

Comment:
I don't think we need to include "constant pool" in this method name; I feel like that is an implementation detail. I would suggest `listExternalMethodReferences`.

Reply:
Yes, I'll update the name.